cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/BufferingTable.java

```diff
@@ -249,46 +250,39 @@ public void startTx(Transaction tx) {
 
   @Override
   public Collection<byte[]> getTxChanges() {
-    switch (conflictLevel) {
-      case NONE:
-        return Collections.emptyList();
-      case ROW:
-        return getRowChanges();
-      case COLUMN:
-        return getColumnChanges();
-      default:
-        throw new RuntimeException("Unknown conflict detection level: " + conflictLevel);
+    if (ConflictDetection.NONE == conflictLevel) {
+      return Collections.emptyList();
     }
-  }
-
-  private Collection<byte[]> getRowChanges() {
-    // we resolve conflicts on row level of individual table
-    List<byte[]> changes = new ArrayList<byte[]>(buff.size());
-    for (byte[] changedRow : buff.keySet()) {
-      changes.add(Bytes.add(getNameAsTxChangePrefix(), changedRow));
+    Collection<byte[]> changes = new TreeSet<byte[]>(UnsignedBytes.lexicographicalComparator());
+    for (Map.Entry<byte[], NavigableMap<byte[], Update>> rowChange : buff.entrySet()) {
+      for (Map.Entry<byte[], Update> updateEntry : rowChange.getValue().entrySet()) {
+        byte[] changeKey = getChangeKey(rowChange.getKey(), updateEntry.getKey(), updateEntry.getValue());
+        if (changeKey != null) {
+          changes.add(Bytes.add(getNameAsTxChangePrefix(), changeKey));
+        }
+      }
     }
     return changes;
   }
 
-  private Collection<byte[]> getColumnChanges() {
-    // we resolve conflicts on row level of individual table
-    List<byte[]> changes = new ArrayList<byte[]>(buff.size());
-    for (Map.Entry<byte[], NavigableMap<byte[], Update>> rowChange : buff.entrySet()) {
-      if (rowChange.getValue() == null) {
-        // NOTE: as of now we cannot detect conflict between delete whole row and row's column value change.
```

Comment:
Interesting note gets deleted... Tephra recently added support for row deletes (are we using that yet?). How does conflict detection work in that case? When conflict detection is at column level but the delete is at row level, how do we detect conflicts?